pypowsybl/dynamic.py

```diff
@@ -0,0 +1,190 @@
+        The dataframe must contains these three columns :
+            - static_id: id of the network element to map (or the id of the generator for omega_ref)
+            - parameter_set_id: set id in the parameter file
+            - mapping_type: value of enum DynamicMappingType
+        For current_limit_automaton the branch side defaults to ONE, see add_current_limit_automaton to add a model with another branch side.
+        For omega_ref the static_id column refers to the generator's id, and the parameter_set_id will be ignored
+        """
+        metadata = _pp.get_dynamic_mappings_meta_data()
+        if kwargs:
+            kwargs = _add_index_to_kwargs(metadata, **kwargs)
+        df = _adapt_df_or_kwargs(metadata, df, **kwargs)
+        c_df = _create_c_dataframe(df, metadata)
+        _pp.add_all_dynamic_mappings(self._handle, c_df)
+
+
+class CurveMapping:
+    def __init__(self) -> None:
+        self._handle = _pp.create_timeseries_mapping()
+
+    def add_curve(self, dynamic_id: str, variable: str) -> None:
+        _pp.add_curve(self._handle, dynamic_id, variable)
+
+    def add_curves(self, dynamic_id: str, variables: _List[str]) -> None:
+        for var in variables:
+            _pp.add_curve(self._handle, dynamic_id, var)
+
+
+class EventType(_Enum):
+    QUADRIPOLE_DISCONNECT = 'QUADRIPOLE_DISCONNECT'
+    SET_POINT_BOOLEAN = 'SET_POINT_BOOLEAN'
+    BRANCH_DISCONNECTION = 'BRANCH_DISCONNECTION'
+
+
+class EventMapping:
+    def __init__(self) -> None:
+        self._handle = _pp.create_event_mapping()
+
+    def add_quadripole_disconnection(self, event_model_id: str, static_id: str, parameter_set_id: str) -> None:
+        _pp.add_event_quadripole_disconnection(
+            self._handle, event_model_id, static_id, parameter_set_id)
+
+    def add_set_point_boolean(self, event_model_id: str, static_id: str, parameter_set_id: str) -> None:
+        _pp.add_event_set_point_boolean(
+            self._handle, event_model_id, static_id, parameter_set_id)
+
+    def add_event(self, parameter_set_id: str, event: EventType, static_id: str, event_id: str = None) -> None:
+        if not event_id:
+            event_id = str(uuid4())
+
+        if event is EventType.QUADRIPOLE_DISCONNECT:
+            _pp.add_event_quadripole_disconnection(
+                self._handle, event_id, static_id, parameter_set_id)
+            return
+        if event is EventType.SET_POINT_BOOLEAN:
+            _pp.add_event_set_point_boolean(
+                self._handle, event_id, static_id, parameter_set_id)
+            return
+
+        raise Exception(
+            f"Pypowsybl-DynamicSimulationError: Unknown event {event}")
+
+    @staticmethod
+    def get_possible_events() -> _List[EventType]:
+        return list(EventType)
+
+
+class SimulationResult:
+    def __init__(self, handle: _pp.JavaHandle) -> None:
+        self._handle = handle
```

Comment:
I propose to initialize all python data from the java handle (status, curves) in the constructor, this will avoid to call the java part each time we want to get the curves, returning a new python object.

Note the difference with the network `get_` methods, because the network is mutable. Here the result is immutable, so it makes more sense not to have `get_` methods.

Reply:
I fixed it. I could not find a good way to make only one java call to get all curves. In Python it is easy to concatenate dataframes, however it is not in java. And there was no easy way to return an array of dataframes without getting into memory allocations troubles. I don't know if I missed an example of this type (passing an array of a generic object from java to python).